src/Compilers/Core/AnalyzerDriver/AnalyzerExecutor.cs

```diff
@@ -529,6 +529,26 @@ internal static bool IsAnalyzerExceptionDiagnostic(Diagnostic diagnostic)
             return false;
         }
 
+        internal static bool AreEquivalentAnalyzerExceptionDiagnostics(Diagnostic exceptionDiagnostic, Diagnostic other)
+        {
+            // We need to have custom de-duplication logic for diagnostics generated for analyzer exceptions.
+            // We create a new descriptor instance per each analyzer exception diagnostic instance (see comments in method "GetAnalyzerExceptionDiagnostic" above).
+            // This is primarily to allow us to embed exception stack trace in the diagnostic description.
+            // However, this might mean that two exception diagnostics which are equivalent in terms of ID and Message, might not have equal description strings.
+            // We want to classify such diagnostics as equal for de-duplication purpose to reduce the noise in output.
```

Comment:
Ah my bad. You are correct, we are not accessing the description field today. So we would only need to change if we decide to include description in the set of fields that we would compare in the future. Please ignore this.